tests/test_gradient.py

```diff
@@ -51,6 +53,71 @@ def test_gradient_checkpointing(self, shape, kernel, space_order):
         gradient2, _ = wave.jacobian_adjoint(residual, u0, vp=v0, checkpointing=False)
         assert np.allclose(gradient.data, gradient2.data)
 
+    @pytest.mark.parametrize('space_order', [4])
+    @pytest.mark.parametrize('kernel', ['OT2'])
+    @pytest.mark.parametrize('shape', [(70, 80)])
```

Comment:
that's quite small, may wanna go with something like `(101, 101)` and `tn=1500)

Reply:
Changed to these values. Test is still not passing though.